hpx/runtime/serialization/detail/polymorphic_nonintrusive_factory.hpp

```diff
@@ -53,13 +53,36 @@ namespace hpx { namespace serialization { namespace detail
     {
         typedef void (*save_function_type) (output_archive& , const void* base);
         typedef void (*load_function_type) (input_archive& , void* base);
-        typedef void* (*create_function_type) ();
+        typedef void* (*create_function_type) (input_archive&);
 
         save_function_type save_function;
         load_function_type load_function;
         create_function_type create_function;
     };
 
+    template <class T>
+    class constructor_selector
+    {
+    public:
+        static T *create(input_archive& ar)
+        {
+            T *t = new T;
+            load_polymorphic(t, ar, hpx::traits::is_nonintrusive_polymorphic<T>());
```

Comment:
This probably needs to be guarded by a try/catch block to avoid potential memory leaks when an exception is thrown.